src/acl/acl.internal.ts

```diff
@@ -491,3 +503,145 @@ export function internal_setAcl<ResourceExt extends WithServerResourceInfo>(
 ): ResourceExt & WithAcl {
   return Object.assign(resource, { internal_acl: acl });
 }
+
+const supportedActorPredicates = [
+  acl.agent,
+  acl.agentClass,
+  acl.agentGroup,
+  acl.origin,
+];
+/**
+ * Union type of all relations defined in `knownActorRelations`.
+ *
+ * When the ACP spec evolves to support additional relations of Rules to Actors,
+ * adding those relations to `knownActorRelations` will cause TypeScript to warn
+ * us everywhere to update everywhere the ActorRelation type is used and that
+ * needs additional work to handle it.
+ */
+type SupportedActorPredicate = typeof supportedActorPredicates extends Array<
+  infer E
+>
+  ? E
+  : never;
+
+/**
+ * Given an ACL Rule, returns two new ACL Rules that cover all the input Rule's use cases,
+ * except for giving the given Agent access to the given Resource.
```

Comment:
Check all comments related to these functions - there's a bunch of references to Agents still :)

```suggestion
 * except for giving the given actor access to the given Resource.
```

Reply:
236d3b7993b68ddd872ec6edacb6b5e70e0281f7